src/Api/ErrorParser/XmlErrorParser.php

```diff
@@ -38,7 +40,17 @@ public function __invoke(
         ];
 
         $body = $response->getBody();
-        if ($body->getSize() > 0) {
+        if (!$body->isSeekable()) {
+            $tempBody = Utils::streamFor();
+            Utils::copyToStream($body, $tempBody);
+            $body = $tempBody;
```

Comment:
this could be (lazy rather than eager):

```php
if (!$body->isSeekable()) {
    $body = new CachingStream($body);
}

Reply:
I don't think it is necessary to have a lazy stream there. The whole content is consumed right after this block, at line 52. Unless you are thinking in any other use case?